diff --git a/tools/lkl/Makefile b/tools/lkl/Makefile
index 777f97e8a..3fc44db37 100644
--- a/tools/lkl/Makefile
+++ b/tools/lkl/Makefile
@@ -96,8 +96,8 @@ $(OUTPUT)liblkl.a: $(OUTPUT)lib/liblkl-in.o $(OUTPUT)lib/lkl.o
 	$(QUIET_AR)$(AR) -rc $@ $^
 
 # rule to link programs
-$(OUTPUT)%$(EXESUF): $(OUTPUT)%-in.o $(OUTPUT)liblkl.a
-	$(QUIET_LINK)$(CC) $(LDFLAGS) $(LDFLAGS_$*-y) -o $@ $^ $(LDLIBS) $(LDLIBS_$*-y)
+$(OUTPUT)%$(EXESUF): $(OUTPUT)%-in.o $(OUTPUT)liblkl$(SOSUF)
+	$(QUIET_LINK)$(CC) $(LDFLAGS) -shared $(LDFLAGS_$*-y) -L$(OUTPUT) -o $@ $< $(LDLIBS) $(LDLIBS_$*-y) -llkl
 
 # rule to build objects
 $(OUTPUT)%-in.o: $(OUTPUT)lib/lkl.o FORCE
